### ISSUE TITLE
Owned entity is creating separate table

### ISSUE THREAD

**Issue by attiqeurrehman** [dotnet/efcore#21601]
Hi,

I am in the process of migrating from .NET Core 2.2 to .NET Core 3.1 but I am having some issues with EF Core 3.1. I have class Widget that owns type Criteria but when I try to query it now it gives me the exception invalid object name 'Widgets1'. I ran the add-migration and it seems to be generating a separate table for the Criteria (which I don't want).

### Steps to reproduce

``` C#
public class Widget 
    {
        public string DashboardId { get; set; }
        public virtual Dashboard Dashboard { get; set; }
        public string Title { get; set; }
        public Criteria Criteria { get; set; }
        public string Type { get; set; }
    }

public class Criteria
    {
        public Resolution Resolution { get; set; }
        public Horizon HorizonOption { get; set; }
        public int DecimalPlaces { get; set; }
        public bool IsMeasurementAllowed { get; set; }
        public bool IsOnlineMode { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
        public decimal? History { get; set; }
        public decimal? Horizon { get; set; }
}

//Entity Configuration 
builder.OwnsOne(w => w.Criteria, wc =>
            {
                wc.Property(c => c.Resolution)
                    .HasConversion(
                    p => p.Name,
                    p => Resolution.FromName(p, false));

                wc.Property(c => c.HorizonOption)
                    .HasConversion(
                    p => p.Name,
                    p => Horizon.FromName(p, false));
            });
```

Got the following exception: 

> 
> Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid object name 'Widgets1'.
>    at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__164_0(Task`1 result)
>    at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
>    at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
>    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
>    at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
> --- End of stack trace from previous location where exception was thrown ---
>    at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
>    at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
>    at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
>    at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.AsyncEnumerator.InitializeReaderAsync(DbContext _, Boolean result, CancellationToken cancellationToken)
>    at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementationAsync[TState,TResult](Func`4 operation, Func`4 verifySucceeded, TState state, CancellationToken cancellationToken)
>    at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementationAsync[TState,TResult](Func`4 operation, Func`4 verifySucceeded, TState state, CancellationToken cancellationToken)
>    at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
>    at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
>    at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)

### Further technical details

EF Core version: 3.1.5
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.1
Operating system: Windows 10
IDE: Visual Studio 2019 16.6.3

**Comment by ajcvickers** [comment]
Please attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing so that we can investigate.

**Comment by AndriySvyryd** [comment]
Likely already fixed in master

**Comment by attiqeurrehman** [comment]
Please find attached sample [OwnedWidgetTypes.zip](https://github.com/dotnet/efcore/files/4919009/OwnedWidgetTypes.zip). Check the migration generated it has table Widget1.

**Comment by ajcvickers** [comment]
Verified fixed with latest daily build. @AndriySvyryd is this a likely dupe?

```C#
public partial class One : Migration
{
    protected override void Up(MigrationBuilder migrationBuilder)
    {
        migrationBuilder.CreateTable(
            name: "Widgets",
            columns: table => new
            {
                Id = table.Column<Guid>(type: "uniqueidentifier", nullable: false),
                CreatedAt = table.Column<DateTime>(type: "datetime2", nullable: false),
                CreatedBy = table.Column<string>(type: "nvarchar(max)", nullable: true),
                UpdatedAt = table.Column<DateTime>(type: "datetime2", nullable: false),
                UpdatedBy = table.Column<string>(type: "nvarchar(max)", nullable: true),
                DashboardId = table.Column<Guid>(type: "uniqueidentifier", nullable: false),
                Title = table.Column<string>(type: "nvarchar(max)", nullable: false),
                Criteria_Resolution = table.Column<string>(type: "nvarchar(max)", nullable: true),
                Criteria_HorizonOption = table.Column<string>(type: "nvarchar(max)", nullable: true),
                Criteria_DecimalPlaces = table.Column<int>(type: "int", nullable: true),
                Criteria_IsMeasurementAllowed = table.Column<bool>(type: "bit", nullable: true),
                Criteria_IsOnlineMode = table.Column<bool>(type: "bit", nullable: true),
                Criteria_FromDate = table.Column<DateTime>(type: "datetime2", nullable: true),
                Criteria_ToDate = table.Column<DateTime>(type: "datetime2", nullable: true),
                Criteria_History = table.Column<decimal>(type: "decimal(18,2)", nullable: true),
                Criteria_Horizon = table.Column<decimal>(type: "decimal(18,2)", nullable: true),
                Type = table.Column<string>(type: "nvarchar(max)", nullable: false),
                ShowLegend = table.Column<bool>(type: "bit", nullable: true),
                XAxisTitle = table.Column<string>(type: "nvarchar(max)", nullable: true),
                YAxisTitle = table.Column<string>(type: "nvarchar(max)", nullable: true),
                SeriesInfos = table.Column<string>(type: "nvarchar(max)", nullable: true),
                Info = table.Column<string>(type: "nvarchar(max)", nullable: true),
                Columns = table.Column<string>(type: "nvarchar(max)", nullable: true)
            },
            constraints: table =>
            {
                table.PrimaryKey("PK_Widgets", x => x.Id);
            });

        migrationBuilder.CreateIndex(
            name: "IX_Widgets_DashboardId",
            table: "Widgets",
            column: "DashboardId");
    }

    protected override void Down(MigrationBuilder migrationBuilder)
    {
        migrationBuilder.DropTable(
            name: "Widgets");
    }
}
```

**Comment by AndriySvyryd** [comment]
@ajcvickers Many issues like this were fixed in https://github.com/dotnet/efcore/pull/19896

**Comment by attiqeurrehman** [comment]
I have updated the sample project to include the version 3.1.6 daily build but it's still looking for the Widget1.

**Comment by ajcvickers** [comment]
@attiqeurrehman This is fixed for the EF Core 5.0 release--you can use the [latest preview](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/5.0.0-preview.6.20312.4) to try it. It's unlikely that this fix will be ported back to EF Core 3.1.x--see [release planning](https://docs.microsoft.com/en-us/ef/core/what-is-new/release-planning) for more details.
